src/game/client/components/menus.cpp

```diff
@@ -1252,19 +1263,26 @@ void CMenus::RenderLoading()
 	Graphics()->BlendNormal();
 	RenderTools()->DrawRoundRect(&Rect, vec4(0.0f, 0.0f, 0.0f, 0.5f), 40.0f);
 
-	const char *pCaption = Localize("Loading");
-
-	Rect.x = x;
-	Rect.y = y+20;
-	Rect.w = w;
-	Rect.h = h;
-	UI()->DoLabel(&Rect, pCaption, 48.0f, CUI::ALIGN_CENTER);
-
-	Rect.x = x+40.0f;
-	Rect.y = y+h-75.0f;
-	Rect.w = (w-80.0f)*Percent;
-	Rect.h = 25.0f;
-	RenderTools()->DrawRoundRect(&Rect, vec4(1.0f, 1.0f, 1.0f, 0.75f), 5.0f);
+	Rect.y += 20;
+	TextRender()->TextColor(1.0f, 1.0f, 1.0f, 1.0f);
+	UI()->DoLabel(&Rect, "Teeworlds", 48.0f, CUI::ALIGN_CENTER);
+
+	float Percent = m_LoadCurrent/(float)m_LoadTotal;
+	float Spacing = 40.0f;
+	float Rounding = 5.0f;
+	CUIRect FullBar = {x+Spacing, y+h-75.0f, w-2*Spacing, 25.0f};
+	RenderTools()->DrawRoundRect(&FullBar, vec4(1.0f, 1.0f, 1.0f, 0.10f), Rounding);
+	CUIRect FillingBar = FullBar;
+	FillingBar.w = (FullBar.w-2*Rounding)*Percent+2*Rounding;
+	RenderTools()->DrawRoundRect(&FillingBar, vec4(1.0f, 1.0f, 1.0f, 0.75f), Rounding);
+
+	if(Percent > 0.5)
+		TextRender()->TextColor(0.2f, 0.2f, 0.2f, 1.0f);
```

Comment:
Nitpicking again, but it would be nice to put a white outline (`TextOutline`) on black text (instead of black on black). At the moment black looks fatter because of that as far as I can tell.